File: exchange_rate_app_react/src/model/ForecastRequestBody.ts
import ForecastFormValue from "./ForecastFormValue";
import moment from "moment";

export class ForecastRequestBody {
    method: string;
    baseCurrency: string;
    forecastCurrency: string;
    startDate: string;
    endDate: string;

    constructor(values: ForecastFormValue) {
        this.method = values.method;
        this.baseCurrency = values.baseCurrency;
        this.forecastCurrency = values.forecastCurrency;
        // const startDateValue = new Date();
        // //Sets minimum date to today
        // startDateValue.setDate(startDateValue.getDate());
        this.startDate = moment().format('YYYY-MM-DD')
        this.endDate = values.date;
    }

}

File: exchange_rate_app_springboot/src/main/java/com/gibatekpro/exchange_rate_app_springboot/model/ForecastRequestBody.java
package com.gibatekpro.exchange_rate_app_springboot.model;

import lombok.Data;

import java.time.LocalDate;

@Data
public class ForecastRequestBody {

    private String method;

    private String baseCurrency;

    private String forecastCurrency;

    private LocalDate startDate;

    private LocalDate endDate;


}
